auth/store.go

```diff
@@ -76,10 +77,24 @@ type AuthStore interface {
 
 	// RoleGrant grants a permission to a role
 	RoleGrant(r *pb.AuthRoleGrantRequest) (*pb.AuthRoleGrantResponse, error)
+
+	// UsernameFromToken gets a username from the given Token
+	UsernameFromToken(token string) (string, bool)
+
+	// IsPutPermitted checks put permission of the user
+	IsPutPermitted(header *pb.RequestHeader, key string) bool
+
+	// IsRangePermitted checks range permission of the user
+	IsRangePermitted(header *pb.RequestHeader, key string) bool
+
+	// IsAuthEnabled checks authentication is enabled or not
+	IsAuthEnabled() bool
```

Comment:
Actually the IsAuthEnabled() isn't used from external packages. Removing the function from the interface and make it private would be better.